doc/api/url.md

```diff
@@ -250,7 +250,463 @@ properties of URL objects:
+
+#### url.search
+
+Gets and sets the serialized query portion of the URL.
+
+```js
+const myURL = new URL('https://example.org/abc?123');
+console.log(myURL.search);
+  // Prints ?123
+
+myURL.search = 'abc=xyz';
+console.log(myURL.href);
+  // Prints https://example.org/abc?abc=xyz
+```
+
+Any invalid URL characters appearing in the value assigned the `search`
+property will be [percent-encoded](#whatwg-percent-encoding). Note that the
+selection of which characters to percent-encode may vary somewhat from what the
+[`url.parse()`][] and [`url.format()`][] methods would produce.
+
+#### url.searchParams
+
+Gets a [`URLSearchParams`](#url_class_urlsearchparams) object representing the
+query parameters of the URL.
+
+#### url.username
+
+Gets and sets the username portion of the URL.
+
+```js
+const myURL = new URL('https://abc:xyz@example.com');
+console.log(myURL.username);
+  // Prints abc
+
+myURL.username = '123';
+console.log(myURL.href);
+  // Prints https://123:xyz@example.com
+```
+
+Any invalid URL characters appearing in the value assigned the `username`
+property will be [percent-encoded](#whatwg-percent-encoding). Note that the
+selection of which characters to percent-encode may vary somewhat from what the
+[`url.parse()`][] and [`url.format()`][] methods would produce.
+
+#### url.toString()
+
+The `toString()` method on the `URL` object returns the serialized URL. The
+value returned is equivalent to that of `url.href`.
+
+### Class: URLSearchParams
+
+The `URLSearchParams` object provides read and write access to the query of a
+`URL`.
+
+```js
+const URL = require('url').URL;
+const myURL = new URL('https://example.org/?abc=123');
+console.log(myURL.searchParams.get('abc'));
+  // Prints 123
+
+myURL.searchParams.append('abc', 'xyz');
+console.log(myURL.href);
+  // Prints https://example.org/?abc=123&abc=xyz
+
+myURL.searchParams.delete('abc');
+myURL.searchParams.set('a', 'b');
+console.log(myURL.href);
+  // Prints https://example.org/?a=b
+```
+
+#### Constructor: new URLSearchParams([init])
+
+* `init` {String} The URL query
+
+#### urlSearchParams.append(name, value)
+
+* `name` {String}
+* `value` {String}
+
+Append a new name-value pair to the query string.
+
+#### urlSearchParams.delete(name)
+
+* `name` {String}
+
+Remove all name-value pairs whose name is `name`.
+
+#### urlSearchParams.entries()
+
+* Returns: {Iterator}
+
+Returns an ES6 Iterator over each of the name-value pairs in the query.
+Each item of the iterator is a JavaScript Array. The first item of the Array
+is the `name`, the second item of the Array is the `value`.
+
+Alias for `urlSearchParams\[\@\@iterator\]()`.
+
+#### urlSearchParams.forEach(fn)
+
+* `fn` {Function} Function invoked for each name-value pair in the query.
+
+Iterates over each name-value pair in the query and invokes the given function.
+
+```js
+const URL = require('url').URL;
+const myURL = new URL('https://example.org/?a=b&c=d');
+myURL.searchParams.forEach((name, value) => {
```

Comment:
It's actually `(value, name)`. Yeah it's counterintuitive but its to keep up with Array and Map's behavior.